node/impl/full/eth.go

```diff
@@ -370,9 +367,17 @@ func (a *EthModule) EthGetTransactionByHashLimited(ctx context.Context, txHash *
 		return nil, nil
 	}
 
-	c, err := a.EthTxHashManager.GetCidFromHash(*txHash)
```

Comment:
Before ChainIndexer this always got enabled if you `EnableEthRPC` (see my note in config/types.go). Now, if you turn on `EnableEthRPC` but don't turn on ChainIndexer then you're going to hit errors in all of these APIs that need to do the look-up (5 or 6 of them I think). We should resolve this, some options:

1. Always enable ChainIndexer when you turn on `EnableEthRPC`
2. Only turn on txhash<>msgcid indexing in ChainIndexer if `EnableEthRPC` but ChainIndexer is not enabled
3. This janky half-way state which is going to have lots of "can't find it" errors for users of the API
4. More explicit error messaging that would help someone figure out they need to enable it

Reply:
I think I'm OK with requiring ChainIndexer if you want Eth RPC. But maybe we shouldn't do the automagical thing like we do now and make it explicit. "Oh, you started the node with `EnableEthRPC` but didn't enable ChainIndexer, that's not going to work". If we make it magical then it's much harder to document. Instead, it's "If you `EnableEthRPC` you must also enable `ChainIndexer` or your node won't start.

Reply:
@rvagg I agree with you. Have fixed this in https://github.com/filecoin-project/lotus/pull/12421/commits/f628cb2ac04ec539f3d31ee6ffb6034c79aed156.

We error out if `ETHRPC` is enabled but `ChainIndexer` is disabled.

Reply:
do you think then we should also error in the `if a.ChainIndexer != nil` cases in this file?

Reply:
@rvagg  What do you mean ?

Reply:
3 functions have `if a.ChainIndexer != nil {` in them, but you've now made it strictly necessary to have a `ChainIndexer` set on start-up. So how about we flip each of those and `if a.ChainIndexer == nil { return nil, errors.New("nope, you configured it bad my dude") }`